clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -3497,10 +3497,10 @@ class SyclKernelBodyCreator : public SyclKernelFieldHandler {
 
   bool enterArray(FieldDecl *FD, QualType ArrayType,
                   QualType ElementType) final {
-    uint64_t ArraySize = SemaRef.getASTContext()
-                             .getAsConstantArrayType(ArrayType)
-                             ->getSize()
-                             .getZExtValue();
+    const ConstantArrayType *CAT =
+        SemaRef.getASTContext().getAsConstantArrayType(ArrayType);
+    assert(CAT && "Should only be called on constant-size array.");
```

Comment:
I think the same check is performed upper the callstack at `visitComplexArray`. Should we check it again?

Reply:
Yea. This path is only entered through visitComplexArray. I don't think we need this check here. Was there some issue?

Reply:
Static analyzers usually have concerns like that.